### ISSUE TITLE
tortoise: rerun is too slow to be executed synchronously

### ISSUE THREAD

**Issue by dshulyak** [spacemeshos/go-spacemesh#2852]
## Description

On devnet205 rerun was started when mesh reached ~3000 layers. In 3h rerun made progress up to ~2000 layer. 

It is executed synchronously while holding a ThreadSafeVerifyingTortoise mutex. It means that for the whole rerun duration hare's outputCollectionLoop is blocked. As a result, the network is stuck for the whole rerun duration.

## Expected Behavior
Rerun is not blocking a network.

**Comment by countvonzero** [comment]
the frequency is decided by `--tortoise-rerun-interval` currently set to 7 days in devnet.
i asked to change it from the original value 1 year. the intention was to flush out the issues and decide what the next step should be. 

my view is that this is a good safety net for bugs in the codebase and should be up and running correctly. but if people feel differently, i guess we can reset it back to 1yr. tho i feel that code doesn't get executed ever should simply be removed.

**Comment by dshulyak** [comment]
It is not usable in the current state, if it takes ~5 hours after 7 days, it will take around a day after a month. Clearly can't block network for that long. It needs to be moved into the background, executed without lock and swap with the original tortoise at some point. 

However, another problem is that it doubles memory consumption. The pointer to the original tortoise state is released only after rerun completes, so effectively the heap size got doubled. It shouldn't be a problem by itself but if there is another issue in place (like with metrics in last devnet) the nodes will simply die.

I agree that not executed code can be as well removed, but i don't know what is the best way to proceed here. I guess it just needs to be a high-priority issue to fix.

**Comment by dshulyak** [comment]
Forgot to close it with the GH.
